src/pkg/task/dao/task.go

```diff
@@ -216,16 +217,43 @@ func (t *taskDAO) querySetter(ctx context.Context, query *q.Query) (orm.QuerySet
 
 	// append the filter for "extra attrs"
 	if query != nil && len(query.Keywords) > 0 {
-		for key, value := range query.Keywords {
+		var (
+			key       string
+			keyPrefix string
+			value     interface{}
+		)
+		for key, value = range query.Keywords {
 			if strings.HasPrefix(key, "ExtraAttrs.") {
-				qs = qs.FilterRaw("id", fmt.Sprintf("in (select id from task where extra_attrs->>'%s'='%s')", strings.TrimPrefix(key, "ExtraAttrs."), value))
+				keyPrefix = "ExtraAttrs."
 				break
 			}
 			if strings.HasPrefix(key, "extra_attrs.") {
-				qs = qs.FilterRaw("id", fmt.Sprintf("in (select id from task where extra_attrs->>'%s'='%s')", strings.TrimPrefix(key, "extra_attrs."), value))
+				keyPrefix = "extra_attrs."
 				break
 			}
 		}
+		if len(keyPrefix) == 0 {
+			return qs, nil
+		}
+		ormer, err := orm.FromContext(ctx)
+		if err != nil {
+			return nil, err
+		}
+		ids := []int64{}
+		// use query parameters rather than string concatenation to avoid the sql injection
+		if _, err = ormer.Raw("select id from task where extra_attrs->>? = ?", strings.TrimPrefix(key, keyPrefix), value).
+			QueryRows(&ids); err != nil {
+			return nil, err
+		}
+		if len(ids) == 0 {
+			ids = append(ids, -1)
+		}
+
+		idStrs := []string{}
+		for _, id := range ids {
+			idStrs = append(idStrs, strconv.FormatInt(id, 10))
+		}
+		qs = qs.FilterRaw("id", fmt.Sprintf("in (%s)", strings.Join(idStrs, ",")))
```

Comment:
This result of `strings.Join(idStrs, ",")` may exceed the max values support by the postgres.

The gc task has the same problem which will be fixed by https://github.com/goharbor/harbor/pull/14195

Reply:
It is in theory. 
But the real case it that there isn't many ids after the filtering and this PR imports no new issues.
We may open a new ticket to track the exceeding max value issue and provide a more general solution, it isn't for task only